test/e2e/certificates_test.go

```diff
@@ -0,0 +1,246 @@
+// +build e2e
+
+package e2e
+
+import (
+	"encoding/base64"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/provider"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/certificates"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/exec"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/namespace"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/openssl"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/pod"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/secret"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/secretproviderclass"
+
+	. "github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1"
+
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("When fetching certificates and private key from Key Vault", func() {
+	var (
+		specName             = "certificates"
+		spc                  *v1alpha1.SecretProviderClass
+		ns                   *corev1.Namespace
+		nodePublishSecretRef *corev1.Secret
+		p                    *corev1.Pod
+	)
+
+	BeforeEach(func() {
+		ns = namespace.Create(namespace.CreateInput{
+			Creator: kubeClient,
+			Name:    specName,
+		})
+
+		nodePublishSecretRef = secret.Create(secret.CreateInput{
+			Creator:   kubeClient,
+			Config:    config,
+			Name:      "secrets-store-creds",
+			Namespace: ns.Name,
+			Data:      map[string][]byte{"clientid": []byte(config.AzureClientID), "clientsecret": []byte(config.AzureClientSecret)},
+		})
+
+		keyVaultObjects := []provider.KeyVaultObject{
+			{
+				ObjectName: "pemcert1",
+				ObjectType: provider.VaultObjectTypeCertificate,
+			},
+			{
+				ObjectName: "pkcs12cert1",
+				ObjectType: provider.VaultObjectTypeCertificate,
+			},
+			{
+				ObjectName: "ecccert1",
+				ObjectType: provider.VaultObjectTypeCertificate,
+			},
+			{
+				ObjectName:  "pemcert1",
+				ObjectType:  provider.VaultObjectTypeKey,
+				ObjectAlias: "pemcert1-pub-key",
+			},
```

Comment:
wow this is much cleaner with Go vs bash

Reply:
yeah 😄